### PR TITLE
Rewrite _onedal_gpu_supported method for SVC

### PR DESCRIPTION
**Pull request by olegkkruglov**
Changes proposed in this pull request:
- Make _onedal_gpu_supported method for SVC shorter and more readable

### CODE REVIEW

**Comment by samir-nasibli** [comment]
@olegkkruglov please resolve the conflict

**Comment by samir-nasibli** [comment]
@please rebase your branches and run intelci

**Comment by samir-nasibli** [comment]
@olegkkruglov please rebase your branches and run intelci

File: sklearnex/svm/svc.py [hunk 1]
```diff
@@ -232,38 +232,25 @@ def _onedal_gpu_supported(self, method_name, *data):
...
-            patching_status.and_conditions(
-                [
-                    (
-                        hasattr(self, "_onedal_estimator")
-                        and self._onedal_gpu_supported("fit", *data),
-                        "oneDAL model was not trained on GPU.",
-                    )
-                ]
+            conditions.append(
+                (hasattr(self, "_onedal_estimator"), "oneDAL model was not trained")
```
Comment (samir-nasibli) [inline sklearnex/svm/svc.py:250]: wrong logic for `conditions` list. Conditions for inference and for train stages are different, don't mix them

**Review by samir-nasibli — CHANGES_REQUESTED** [review]
Please rewrite it/ make a refactoring without changing the logic.

File: sklearnex/svm/svc.py [hunk 2]
```diff
@@ -232,38 +232,25 @@ def _onedal_gpu_supported(self, method_name, *data):
...
             )
-            return patching_status.get_status(logs=True)
-        raise RuntimeError(f"Unknown method {method_name} in {class_name}")
```
Comment (Alexsandruss) [inline sklearnex/svm/svc.py:266]: RuntimeError should exist at the end of function like in other estimators or this support checking workflow should be changed

**Review by Alexsandruss — CHANGES_REQUESTED** [review]

**Comment by samir-nasibli** [comment]
GPU CI Job: http://intel-ci.intel.com/ee467fd9-23b2-f12c-b9f3-a4bf010d0e2e. Specifically for sklearnex/daal4py tests/examples on GPU.

**Pull request merged**